client/src/Components/RelatedItemsAndComparison/RelatedCards.jsx

```diff
@@ -3,33 +3,61 @@ import axios from 'axios';
 import ReactDOM from 'react-dom';
 import ComparisonModal from './ComparisonModal.jsx'
 import styled from 'styled-components';
+import StarRating from '../SharedComponents/StarRating.jsx';
 
 const RelatedCards = (props) => {
-  const [modal, setModal] = useState(false)
-  const [products, setProducts] = useState([])
+  const [modal, setModal] = useState(false);
+  const [products, setProducts] = useState([]);
+  const [clickedProd, setClickedProd] = useState({});
   var modalToggle = () => {
     setModal(!modal)
   }
+
   useEffect(() => {
-    console.log('index', props.relatedIndex)
-    console.log('slice', props.relatedProd.slice(props.relatedIndex, props.relatedIndex + 4))
-    console.log('related id', props.relatedId)
     setProducts(props.relatedProd.slice(props.relatedIndex, props.relatedIndex + 4));
   }, [props.relatedIndex, props.relatedProd])
+
   async function handleCardClick(id) {
     return props.setProductId(id).then(() => {props.setRelatedIndex(0)})
   }
+
+  const getUrl = (id) => {
+    if (props.styles !== []) {
+      for (var i = 0; i < props.styles.length; i++) {
+        if (props.styles[i].product_id == id) {
+          return props.styles[i].results[0].photos[0].thumbnail_url
+        }
+      }
+    }
+  }
+  const getRatings = (id) => {
+    if (props.reviewData !== []) {
+      for (var i = 0; i < props.reviewData.length; i++) {
+        if (props.reviewData[i].product_id == id) {
+          return props.reviewData[i].ratings
+        }
+      }
+    }
+  }
   return (
     <>
     {products?.map((prod) => {
+      {console.log('url', getUrl(prod.id))}
```

Comment:
The url gets console logged many times upon updating the selected product. Would you mind removing the console log? That is, only if it is unnecessary. I don't know if you want to keep it for future use.